Makefile

```diff
@@ -19,6 +19,7 @@ test_luv:
 	EIO_BACKEND=luv dune runtest
 
 dscheck:
+	dune exec -- ./lib_eio/tests/dscheck/test_sync.exe
```

Comment:
If I run `make dschek` on my mac, it only seems to run the first of these tests:

```sh
➜  eio git:(sync) ✗ make dscheck      
dune exec -- ./lib_eio/tests/dscheck/test_sync.exe
Done: 40% (86/215, 129 left) (jobs: 0)run: 1000
run: 2000
... snip lots of similar lines ...
run: 28880000
rumake: *** [dscheck] Killed: 9
➜  eio git:(sync) ✗ 
```

Reply:
On the same note, running `make bench` fails:

```sh
dune exec -- ./lib_eio_linux/tests/bench_noop.exe
Error: Program "./lib_eio_linux/tests/bench_noop.exe" not found!
make: *** [bench] Error 1         
```

Reply:
Well, the same also goes for `dune build`:

```sh
➜  eio git:(sync) ✗ dune build
File "lib_eio_linux/tests/dune", line 30, characters 21-30:
30 |  (libraries alcotest eio_linux))
                          ^^^^^^^^^
Error: Library "eio_linux" in _build/default/lib_eio_linux is hidden
(unsatisfied 'enabled_if').
-> required by _build/default/lib_eio_linux/tests/test.exe
-> required by alias lib_eio_linux/tests/all
-> required by alias default
Error: The package eio_linux does not have any user defined stanzas attached
to it. If this is intentional, add (allow_empty) to the package definition in
the dune-project file
-> required by _build/default/eio_linux.install
-> required by alias all
-> required by alias default
```

It would be nice that the build wouldn't appear to fail on mac.  I don't know how to fix this immediately, but I can take a look after reviewing this PR.

Reply:
`make dscheck` currently requires a patched version of dscheck, as the official version is too slow to be useful: https://github.com/ocaml-multicore/dscheck/pull/3. However, that version has a bug which means it may also miss some cases.

Ideally, we'd replace the explicit commands in `make bench` and `make dscheck` with e.g. `dune build @dscheck`. Then dune could be smarter about filtering things out on mac. But then you have to stop it running things in parallel, disable buffered output, and force it to let you run the benchmarks multiple times without changes to the code, which is a bit of a pain.

Fixing the main build on mac is important, though! I don't know why dune is trying to build it though. We already have:
```
(test
 (name test)
 (package eio_linux)
 (enabled_if (= %{system} "linux"))
 (modules test)
 (libraries alcotest eio_linux))
```

Reply:
A related issue in dune: [The documentation/value of %{system} is not consistent](https://github.com/ocaml/dune/issues/4895).

I also tried to work around this by using the `(* -*- tuareg -*- *)` magic, but that didn't work either as that hits a limitation in dune: [(dirs ...) not recognised in dune2](https://github.com/ocaml/dune/issues/2938).

Reply:
I've reported it at https://github.com/ocaml/dune/issues/6938